src/table.ts

```diff
@@ -56,11 +56,19 @@ export type JobMetadataResponse = [JobMetadata];
 // tslint:disable-next-line no-any
 export type RowMetadata = any;
 
-export type InsertRowsOptions = bigquery.ITableDataInsertAllRequest&{
-  autoCreate?: boolean;
-  raw?: boolean;
+export type InsertRowsOptionsAutoCreate = {
+  autoCreate?: false;
+  schema?: string | {};
+}|{
+  autoCreate: true;
   schema: string|{};
```

Comment:
Any reason why we don't just make `schema` optional? I like the overloads, but I can't imagine a scenario where `autoCreate` would be set to `false` and `schema` would still be present

Reply:
It doesn't make sense IMO. I followed the documentation as closely as possible. It does not state that this property cannot be set if `autoCreate` is not `true`.

If there are just two possibilities: `autoCreate: true` + `schema` or none of these, why do we even have two properties? Just for readability? In this case, would you prefer to disallow providing the `schema` if `autoCreate` is not `true`?

Reply:
That is a documentation failure on our part then, we completely ignore the `schema` unless `autoCreate: true` is set.

> why do we even have two properties?

That is a good question, `autoCreate` is just a convention used across all the Node.js clients, but as you've pointed out it seems kind of pointless here since `schema` basically implies the same thing. We should open a feature request ticket to merge them into a single option

Reply:
Ok, I can open an issue. I guess there is a chance to make it a non breaking change if we start ignoring `autoCreate` and start relying on `schema`, or introduce yet another property and keep these two for compatibility, at least for some time. I’d be happy to open an issue and work on a PR once we align on a design.

But for the time being, how do we want to approach fixing current type definitions?

Reply:
I'm ok with just marking schema as optional, WDYT?

Reply:
Done :)